### PR TITLE
feat: renamed ensure_none to unsure_unsigned

### LINKED ISSUE #13279: `ensure_none` naming is misleading
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

FRAME System's `ensure_none` ensures that the origin `o` represents an unsigned extrinsic.

This naming is consistent with the inner member of the `RawOrigin` enum:
```
pub enum RawOrigin<AccountId> {
	Root,
	Signed(AccountId),
	None,
}
```

During a Milestone Review for the Substrate Builders Program I had to clarify a team that was misled by the  `ensure_none` naming. The team had interpreted that `ensure_none` meant no one could call this extrinsic, which is literally the opposite of the meaning we're aiming to convey.

Even though [the documentation](https://paritytech.github.io/substrate/master/frame_system/fn.ensure_none.html) already provides an accurate description, we could further avoid confusions by renaming `ensure_none` to `ensure_unsigned`.

### Steps to reproduce

_No response_

### PR DESCRIPTION
**Pull request by gitofdeepanshu**
Fix #13279 
## Description
Deprecate `ensure_none` and introduced the alias name `ensure_unsigned`. 
### NOTE
Not a breaking change

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2343453

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2343452

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2343451

**Review by xlc — CHANGES_REQUESTED** [review]
Please don't just break common API without notice.
You need to kept `ensure_none` alias and mark it as `#[deprecated]`.

**Comment by gitofdeepanshu** [comment]
> Please don't just break common API without notice. You need to kept `ensure_none` alias and mark it as `#[deprecated]`.

Added 'ensure_none' alias as requested by @xlc .

**Comment by gitofdeepanshu** [comment]
btw is it just me or does it take 30 minutes for everyone to build?

**Comment by ggwpez** [comment]
> btw is it just me or does it take 30 minutes for everyone to build?

Building the whole thing can take a while without Sccache and fast hardware. Normally compiling a few singluar crates + CI checks works good. I often just do `cargo c -p pallet-balances` or whatever for small changes.

Please update the description to mention the alias and that this is *not* a breaking change.

**Comment by gitofdeepanshu** [comment]
> Please update the description to mention the alias and that this is _not_ a breaking change.

Done.
Although local CI checks are passing, Polkadot CI is failing. `ensure_none` still exists but Polkadot can't seem to find it.

File: frame/system/src/lib.rs [hunk 1]
```diff
@@ -1753,7 +1764,7 @@ impl<T: Config> Lookup for ChainContext<T> {
 
 /// Prelude to be used alongside pallet macro, for ease of use.
 pub mod pallet_prelude {
-	pub use crate::{ensure_none, ensure_root, ensure_signed, ensure_signed_or_root};
```
Comment (ggwpez) [inline frame/system/src/lib.rs:1756]: You still have to export it, otherwise this is a braking change and Polkadot will not find it anymore.

File: frame/system/src/lib.rs [hunk 2]
```diff
@@ -975,8 +975,19 @@ where
 	}
 }
 
-/// Ensure that the origin `o` represents an unsigned extrinsic. Returns `Ok` or an `Err` otherwise.
+/// Earlier Usage:
+/// Ensure that the origin `o` represents the root. Returns `Ok` or an `Err` otherwise.
+/// Now Deprecated, Use `ensure_unsigned` instead
```
Comment (ggwpez) [inline frame/system/src/lib.rs:980]: I dont get this comment. The function did not change. Why do you now mention `root`?

File: frame/system/src/lib.rs [hunk 3]
```diff
@@ -975,8 +975,19 @@ where
...
+#[deprecated]
```
Comment (ggwpez) [inline frame/system/src/lib.rs:981]:
```suggestion
#[deprecated(note = "Renamed to 'ensure_unsigned' as clarification")]
```

**Comment by davxy** [comment]
IMO the naming now may be more misleading than before.

We have an enum:
```
pub enum RawOrigin<AccountId> {
    Root,
    Signed(AccountId),
    None,
}
```

Currently there is a check function for each variant (where the name of the function is in 1 to 1 correspondence to the variant name)... Pretty easy
- `ensure_signed` <-> `Signed<T>`
- `ensure_root` <-> `Root`
- `ensure_none` <-> `None`

As a user of `ensure_unsigned` I may logically  (wrongly) infer that `ensure_unsigned` is the opposite of `ensure_signed`. 
But it is not. Since Root is not allowed by both of them. 

In short, `ensure_signed(org) || ensure_unsigned(org)` is not a tautology as one may expect.

Obviously reading the docs removes any doubt. But the same applies for the previous implementation

**Review by davxy — COMMENTED** [review]
IMO the naming now may be more misleading than before.

We have an enum:
```
pub enum RawOrigin<AccountId> {
    Root,
    Signed(AccountId),
    None,
}
```

Currently there is a check function for each variant (where the name of the function is in 1 to 1 correspondence to the variant name)... Pretty easy
- `ensure_signed` <-> `Signed<T>`
- `ensure_root` <-> `Root`
- `ensure_none` <-> `None`

As a user of the new `ensure_unsigned` I may logically  (wrongly) infer that `ensure_unsigned` is the opposite of `ensure_signed`. 
But it is not, since `Root` origin is not allowed by both of them. 

In short, `ensure_signed(org) || ensure_unsigned(org)` is not a tautology as one may expect.

Obviously reading the docs removes any doubt. But the same "read the docs" approach applies to the previous implementation as well

**Comment by bkchr** [comment]
> As a user of the new `ensure_unsigned` I may logically (wrongly) infer that `ensure_unsigned` is the opposite of `ensure_signed`.
> But it is not, since `Root` origin is not allowed by both of them.

While this is true, you could not assume that `ensure_none` is meaning "no one can call this". And also nothing bad would be happen if you use `ensure_unsigned` and it is called with `Root`.

I think the general direction is fine, but we should also rename `None` in the enum `RawOrigin` to `Unsigned`. There are also usages of `none` like here: https://github.com/paritytech/substrate/blob/f5e370da7f34622637a343f03420b4242b321891/frame/support/src/traits/dispatch.rs#L349-L350 

So, if we do it, we should do it everywhere and use `deprecated` where possible.

**Comment by juangirini** [comment]
@gitofdeepanshu would you take this issue over?
